soba stopコマンドがエラーになる

douhashi@DHS-OXPX1  bin/soba stop
2025-09-23 15:59:09.704 [ERROR] Repository configuration is required but not provided component=dependency-resolver repository= owner= repo=
2025-09-23 15:59:09.704 [ERROR] Failed to resolve services component=builder error="repository configuration is required:
owner="" repo="" (repository="")"
2025-09-23 15:59:09.704 [ERROR] Failed to create daemon service component=daemon error="failed to resolve services: repository configuration is required: owner="" repo="" (repository="")"
panic: Failed to build daemon service: failed to resolve services: repository configuration is required: owner="" repo=""
(repository="")

goroutine 1 [running]:
github.com/douhashi/soba/internal/service.NewDaemonService(0xc000130360)
        /home/douhashi/workspace/github.com/douhashi/soba/internal/service/daemon.go:67 +0x285
github.com/douhashi/soba/internal/cli.runStop(0xc00012b508, {0xe0be20, 0x0, 0x0})
        /home/douhashi/workspace/github.com/douhashi/soba/internal/cli/stop.go:27 +0x30
github.com/douhashi/soba/internal/cli.newRootCmd.newStopCmd.func4(0xc00012b508?, {0xe0be20?, 0x4?, 0x998e7d?})
        /home/douhashi/workspace/github.com/douhashi/soba/internal/cli/stop.go:19 +0x18
github.com/spf13/cobra.(*Command).execute(0xc00012b508, {0xe0be20, 0x0, 0x0})
        /home/douhashi/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1015 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc00012a308)
        /home/douhashi/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
        /home/douhashi/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
github.com/douhashi/soba/internal/cli.Execute(...)
        /home/douhashi/workspace/github.com/douhashi/soba/internal/cli/root.go:74
main.main()
        /home/douhashi/workspace/github.com/douhashi/soba/cmd/soba/main.go:17 +0xac

## 注意事項
- 保守性の良い最善のアーキテクチャ深く考えて選択するようにしてください。(既存の実装を尊重しすぎる必要はありません)
- 互換性の維持は不要 (後方互換性を維持するためにコードが複雑化するほうが害悪です)
- フルテストがパスすること

--- a/internal/cli/root.go
+++ b/internal/cli/root.go
@@ -45,7 +45,7 @@ development workflows through seamless integration with Claude Code AI.`,
 				cmdName = cmd.Name()
 			}
 
-			if cmdName == "init" || cmdName == "version" {
+			if cmdName == "init" || cmdName == "version" || cmdName == "stop" {
 				return nil
 			}
 

--- a/internal/cli/stop.go
+++ b/internal/cli/stop.go
@@ -24,7 +24,11 @@ func newStopCmd() *cobra.Command {
 }
 
 func runStop(cmd *cobra.Command, args []string) error {
-	daemonService := service.NewDaemonService(app.LogFactory())
+	// Initialize app with minimal setup for stop command
+	// This avoids the need for config file
+	app.MustInitializeMinimal()
+
+	daemonService := service.NewDaemonServiceForStop(app.LogFactory())
 	return runStopWithService(cmd, args, daemonService)
 }
 
@@ -37,13 +41,14 @@ type StopServiceInterface interface {
 func runStopWithService(cmd *cobra.Command, _ []string, daemonService StopServiceInterface) error {
 	log := app.LogFactory().CreateComponentLogger("cli")
 
-	// Get config from global app
-	cfg := app.Config()
-
-	// Get repository from config
+	// Try to get config from global app if available
+	// But don't fail if config is not available
 	var repository string
-	if cfg != nil {
-		repository = cfg.GitHub.Repository
+	if app.IsInitialized() {
+		cfg := app.Config()
+		if cfg != nil {
+			repository = cfg.GitHub.Repository
+		}
 	}
 
 	ctx := context.Background()

--- a/internal/cli/stop_integration_test.go
+++ b/internal/cli/stop_integration_test.go
@@ -11,6 +11,8 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/douhashi/soba/pkg/app"
 )
 
 func TestStopCommandIntegration(t *testing.T) {
@@ -129,6 +131,58 @@ workflow:
 
 		// PIDファイルを手動で削除
 		os.Remove(pidFile)
+	})
+}
+
+func TestStopCommandWithoutConfig(t *testing.T) {
+	// Reset app state for clean test
+	app.Reset()
+
+	// テスト用の一時ディレクトリを作成（設定ファイルなし）
+	tmpDir := t.TempDir()
+	sobaDir := filepath.Join(tmpDir, ".soba")
+	require.NoError(t, os.MkdirAll(sobaDir, 0755))
+
+	// 元のディレクトリを保存して、テスト終了後に戻る
+	originalDir, err := os.Getwd()
+	require.NoError(t, err)
+	defer os.Chdir(originalDir)
+
+	// 作業ディレクトリを変更
+	require.NoError(t, os.Chdir(tmpDir))
+
+	t.Run("Stop without config file when daemon is not running", func(t *testing.T) {
+		// stopコマンドを実行（設定ファイルなし）
+		cmd := newStopCmd()
+		cmd.SetArgs([]string{})
+
+		// デーモンが起動していない状態でstopを実行（エラーが返る）
+		err := cmd.Execute()
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "daemon is not running")
+		// 重要: repository configuration requiredエラーが出ないことを確認
+		assert.NotContains(t, err.Error(), "repository configuration is required")
+	})
+
+	t.Run("Stop without config file with PID file", func(t *testing.T) {
+		// PIDファイルを作成（存在しないPID）
+		pidFile := filepath.Join(sobaDir, "soba.pid")
+		require.NoError(t, os.WriteFile(pidFile, []byte("999999"), 0600))
+
+		// stopコマンドを実行
+		cmd := newStopCmd()
+		cmd.SetArgs([]string{})
+
+		// PIDが存在しないプロセスのため、エラーになる
+		err := cmd.Execute()
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "process not found")
+		// 重要: repository configuration requiredエラーが出ないことを確認
+		assert.NotContains(t, err.Error(), "repository configuration is required")
+
+		// PIDファイルが削除されていることを確認
+		_, err = os.Stat(pidFile)
+		assert.True(t, os.IsNotExist(err), "PID file should be deleted")
 	})
 }
 

--- a/internal/cli/stop_test.go
+++ b/internal/cli/stop_test.go
@@ -22,13 +22,10 @@ func (m *MockStopService) Stop(ctx context.Context, repository string) error {
 }
 
 func TestStopCommand(t *testing.T) {
-	// Initialize app for testing
-	helper := app.NewTestHelper(t)
-	helper.InitializeForTest()
-
 	tests := []struct {
 		name           string
 		args           []string
+		setupApp       func()
 		setupMock      func(*MockStopService)
 		expectedOutput string
 		wantError      bool
@@ -36,6 +33,11 @@ func TestStopCommand(t *testing.T) {
 		{
 			name: "Stop daemon successfully",
 			args: []string{},
+			setupApp: func() {
+				// Initialize app for testing
+				helper := app.NewTestHelper(t)
+				helper.InitializeForTest()
+			},
 			setupMock: func(daemon *MockStopService) {
 				daemon.On("Stop", mock.Anything, mock.Anything).Return(nil)
 			},
@@ -45,16 +47,40 @@ func TestStopCommand(t *testing.T) {
 		{
 			name: "Stop daemon with error",
 			args: []string{},
+			setupApp: func() {
+				// Initialize app for testing
+				helper := app.NewTestHelper(t)
+				helper.InitializeForTest()
+			},
 			setupMock: func(daemon *MockStopService) {
 				daemon.On("Stop", mock.Anything, mock.Anything).Return(assert.AnError)
 			},
 			expectedOutput: "",
 			wantError:      true,
 		},
+		{
+			name: "Stop daemon without config file",
+			args: []string{},
+			setupApp: func() {
+				// Reset app to ensure no config is loaded
+				app.Reset()
+				// Initialize minimal app without config
+				app.MustInitializeMinimal()
+			},
+			setupMock: func(daemon *MockStopService) {
+				// Repository情報が空でも動作することを確認
+				daemon.On("Stop", mock.Anything, "").Return(nil)
+			},
+			expectedOutput: "Daemon stopped successfully\n",
+			wantError:      false,
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			// Setup app state
+			tt.setupApp()
+
 			// モックの準備
 			mockDaemon := new(MockStopService)
 			tt.setupMock(mockDaemon)
@@ -81,6 +107,9 @@ func TestStopCommand(t *testing.T) {
 
 			// モックの検証
 			mockDaemon.AssertExpectations(t)
+
+			// Clean up
+			app.Reset()
 		})
 	}
 }

--- a/internal/service/daemon.go
+++ b/internal/service/daemon.go
@@ -92,6 +92,30 @@ func NewDaemonServiceWithConfig(cfg *config.Config, logFactory *logging.Factory)
 	return service
 }
 
+// NewDaemonServiceForStop creates a minimal daemon service for stop command
+// This version doesn't require config and is used only for stopping the daemon
+func NewDaemonServiceForStop(logFactory *logging.Factory) DaemonService {
+	logger := logFactory.CreateComponentLogger("daemon")
+
+	// Get current working directory
+	workDir, err := os.Getwd()
+	if err != nil {
+		workDir = "."
+	}
+
+	// Create minimal daemon service with only the necessary dependencies
+	// tmux client is optional and will be initialized as nil
+	return &daemonService{
+		workDir:                   workDir,
+		processor:                 nil, // Not needed for stop
+		watcher:                   nil, // Not needed for stop
+		prWatcher:                 nil, // Not needed for stop
+		closedIssueCleanupService: nil, // Not needed for stop
+		tmux:                      nil, // Optional for stop
+		logger:                    logger,
+	}
+}
+
 // NewDaemonServiceWithDependencies creates daemon service with injected dependencies
 func NewDaemonServiceWithDependencies(
 	workDir string,

--- a/internal/service/daemon_test.go
+++ b/internal/service/daemon_test.go
@@ -887,3 +887,47 @@ func TestDaemonService_ClosedIssueCleanupServiceStartupLog(t *testing.T) {
 		})
 	}
 }
+
+// TestNewDaemonServiceForStop tests the Stop-specific daemon service factory
+func TestNewDaemonServiceForStop(t *testing.T) {
+	// Initialize app for testing
+	helper := app.NewTestHelper(t)
+	helper.InitializeForTest()
+
+	// Test that NewDaemonServiceForStop creates a service without config dependency
+	service := NewDaemonServiceForStop(app.LogFactory())
+	assert.NotNil(t, service, "NewDaemonServiceForStop should create a daemon service")
+}
+
+// TestStopMethodWithEmptyRepository tests Stop method with empty repository
+func TestStopMethodWithEmptyRepository(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	// Initialize app for testing
+	helper := app.NewTestHelper(t)
+	helper.InitializeForTest()
+
+	// Create a minimal daemon service for Stop
+	daemonService := &daemonService{
+		workDir: tmpDir,
+		tmux:    nil, // tmux client is optional for Stop
+		logger:  app.LogFactory().CreateComponentLogger("daemon-test"),
+	}
+
+	// Create PID file to simulate running daemon
+	pidFile := filepath.Join(tmpDir, ".soba", "soba.pid")
+	require.NoError(t, os.MkdirAll(filepath.Join(tmpDir, ".soba"), 0755))
+	require.NoError(t, os.WriteFile(pidFile, []byte("99999"), 0644)) // Non-existent PID
+
+	// Test Stop with empty repository
+	ctx := context.Background()
+	err := daemonService.Stop(ctx, "")
+
+	// Should handle gracefully (process not found is expected)
+	assert.Error(t, err, "Should return error for non-existent process")
+	assert.Contains(t, err.Error(), "process not found")
+
+	// PID file should be removed even on error
+	_, statErr := os.Stat(pidFile)
+	assert.True(t, os.IsNotExist(statErr), "PID file should be removed")
+}

--- a/pkg/app/app.go
+++ b/pkg/app/app.go
@@ -74,6 +74,34 @@ func MustInitializeWithOptions(configPath string, opts *InitOptions) {
 	initialized = true
 }
 
+// MustInitializeMinimal initializes minimal app state for commands that don't need config
+func MustInitializeMinimal() {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if initialized {
+		// Already initialized, nothing to do
+		return
+	}
+
+	// Create Logger Factory with defaults
+	var err error
+	logFactory, err = logging.NewFactory(logging.Config{
+		Level:        "warn",
+		Format:       "text",
+		Output:       "stdout",
+		AlsoToStdout: false,
+	})
+	if err != nil {
+		panic("failed to create logger factory: " + err.Error())
+	}
+
+	// cfg remains nil for minimal initialization
+	// Slack is not initialized
+
+	initialized = true
+}
+
 // MustInitializeForTest initializes the application for testing
 func MustInitializeForTest(testConfig *config.Config) {
 	mu.Lock()

